test/e2e/suite/peered_vpc.go

```diff
@@ -0,0 +1,285 @@
+package suite
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awsconfig "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation"
+	ec2v2 "github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	s3v2 "github.com/aws/aws-sdk-go-v2/service/s3"
+	ssmv2 "github.com/aws/aws-sdk-go-v2/service/ssm"
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo/v2"
+	"github.com/onsi/ginkgo/v2/types"
+	. "github.com/onsi/gomega"
+	clientgo "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/aws/eks-hybrid/test/e2e"
+	"github.com/aws/eks-hybrid/test/e2e/addon"
+	"github.com/aws/eks-hybrid/test/e2e/cluster"
+	"github.com/aws/eks-hybrid/test/e2e/commands"
+	"github.com/aws/eks-hybrid/test/e2e/credentials"
+	"github.com/aws/eks-hybrid/test/e2e/kubernetes"
+	"github.com/aws/eks-hybrid/test/e2e/nodeadm"
+	"github.com/aws/eks-hybrid/test/e2e/peered"
+	"github.com/aws/eks-hybrid/test/e2e/s3"
+	"github.com/aws/eks-hybrid/test/e2e/ssm"
+)
+
+type suiteConfiguration struct {
+	TestConfig             *e2e.TestConfig          `json:"testConfig"`
+	SkipCleanup            bool                     `json:"skipCleanup"`
+	CredentialsStackOutput *credentials.StackOutput `json:"ec2StackOutput"`
+	RolesAnywhereCACertPEM []byte                   `json:"rolesAnywhereCACertPEM"`
+	RolesAnywhereCAKeyPEM  []byte                   `json:"rolesAnywhereCAPrivateKeyPEM"`
+	PublicKey              string                   `json:"publicKey"`
+	JumpboxInstanceId      string                   `json:"jumpboxInstanceId"`
+}
+
+type peeredVPCTest struct {
+	aws             aws.Config
+	eksClient       *eks.Client
+	ec2Client       *ec2v2.Client
+	ssmClient       *ssmv2.Client
+	cfnClient       *cloudformation.Client
+	k8sClient       clientgo.Interface
+	k8sClientConfig *rest.Config
+	s3Client        *s3v2.Client
+	iamClient       *iam.Client
+
+	logger        logr.Logger
+	loggerControl e2e.PausableLogger
+	logsBucket    string
+	artifactsPath string
+
+	cluster         *peered.HybridCluster
+	stackOut        *credentials.StackOutput
+	nodeadmURLs     e2e.NodeadmURLs
+	rolesAnywhereCA *credentials.Certificate
+
+	overrideNodeK8sVersion string
+	setRootPassword        bool
+	skipCleanup            bool
+
+	publicKey string
+
+	remoteCommandRunner commands.RemoteCommandRunner
+
+	podIdentityS3Bucket string
+
+	// failureMessageLogged tracks if a terminal error due to a failed gomega
+	// expectation has already been registered and logged . It avoids logging
+	// the same multiple times.
+	failureMessageLogged bool
+}
+
+type verificationProvider interface {
+	newPeeredNode() *peered.Node
+	newVerifyNode(nodeName, nodeIP string) *kubernetes.VerifyNode
```

Comment:
what if we just move these two methods to testNode? it doesn't seem like they are needed here anymore since the test code now access that functionality through the testNode abstraction.